### PR TITLE
update dtos message structure

### PR DESCRIPTION
**Pull request by mbrenguier**

### CODE REVIEW

File: Library/Dtos/Message.php [hunk 1]
```diff
@@ -58,4 +58,8 @@ public function offsetUnset($offset) {
     public function offsetGet($offset) {
```
Comment (jrmbrgs) [inline Library/Dtos/Message.php]: We should remove the implementation of array access, it was just here to make easier the transition between php array to DTO

File: Library/Dtos/Message.php [hunk 2]
```diff
@@ -58,4 +58,8 @@ public function offsetUnset($offset) {
...
         return isset($this->rawHash[$offset]) ? $this->rawHash[$offset] : null;
     }
+
+    public function getPayload() {
```
Comment (jrmbrgs) [inline Library/Dtos/Message.php]: maybe add test if non empty and add the return type to be safer

File: example/Services/StepInterface.php [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<?php
+namespace Ex\Services;
```
Comment (jrmbrgs) [inline example/Services/StepInterface.php]: whey do you redelcare the interface in the example ?
It's supposed to be provided by \Disturb, the client code should not have to re-declare it

File: Library/Step/AbstractStep.php [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?php
+namespace Disturb\Step;
+
+abstract class AbstractStep extends \Phalcon\Mvc\User\Component
```
Comment (jrmbrgs) [inline Library/Step/AbstractStep.php]: Maybe it would be less coercive to expose message returns type as const e.g. \Disturb\Message\Return::Success \Disturb\Message\Return::Failure rather than forcing client to extends the abstract class which provides "poor logic"

**Review by jrmbrgs — CHANGES_REQUESTED** [review]
Some general comments, not directly related to you PR : 
 * we should remove the ...\Dtos\... in the namespace, just \Disturb\Message, class type should not be in namespaces
 * DTO should provide a `isValid()` method
 * keep in mind to make Disturb as independent as  possible, in other words, avoid interfaces to implement or abstract classes to extends client side

File: example/Services/StepInterface.php (hunk 3 above)
Comment (mbrenguier) [reply example/Services/StepInterface.php]: mistake

File: Library/Dtos/Message.php (hunk 1 above)
Comment (jrmbrgs) [reply Library/Dtos/Message.php]: will do it

File: Library/Tasks/ManagerTask.php [hunk 5]
```diff
@@ -59,13 +59,16 @@ protected function processMessage(\Disturb\Dtos\Message $payloadHash)
...
         // run through the next step(s)
-        foreach($stepTaskHashList as $stepTaskHash) {
+        foreach ($stepTaskHashList as $stepTaskHash) {
             $stepCode = $stepTaskHash['name'];
             $stepInputList = $this->service->getStepInput($workflowProcessId, $stepCode);
             // run through the "job" to send to each step
-            foreach($stepInputList as $stepJob) {
-                $stepJob['type'] = \Disturb\Dtos\Message::TYPE_STEP_CTRL;
-                $stepMessageDto = new \Disturb\Dtos\Message(json_encode($stepJob));
+            foreach ($stepInputList as $stepJob) {
```
Comment (jrmbrgs) [inline Library/Tasks/ManagerTask.php]: stepJobHash

**Review by jrmbrgs — CHANGES_REQUESTED** [review]
